### DISCUSSION TITLE
GLB model renders incorrectly

### DISCUSSION THREAD

**Discussion by VisualEhrmanntraut** [bevyengine/bevy#3066] [Q&A] [answered]
In game:
![image](https://user-images.githubusercontent.com/30368284/140412463-930b73cd-476e-455f-8cd1-dfaf4de7166b.png)
In blender:
![image](https://user-images.githubusercontent.com/30368284/140412475-a375629c-0f32-4dfa-9d51-8fcd0a04a4b6.png)
Code:
![image](https://user-images.githubusercontent.com/30368284/140412949-22cc2e30-0904-425d-808c-8588dfbf8665.png)
[parappa.glb.zip](https://github.com/bevyengine/bevy/files/7478208/parappa.glb.zip)
)

**Comment by DJMcNab** [comment]
Thanks for sharing the code, however please provide it as actual code. Markdown supports code blocks, e.g.
````
```rust
fn main(){}
```
````
results in 
```rust
fn main()
```

Please also provide what bevy version you are using. I suspect the problem is some issue of winding direction or backface culling, but I'm not certain.

**Reply by VisualEhrmanntraut** [reply]
Thanks, but, I'm aware, just didn't think it was necessary.
```rust
    commands.insert_resource(audio_state);

    commands.spawn_scene(asset_server.load("models/parappa.glb#Scene0"));

    commands.spawn_bundle(PbrBundle {
        mesh: meshes.add(Mesh::from(shape::Plane { size: 5.0 })),
        material: materials.add(Color::rgb(0.3, 0.5, 0.3).into()),
        ..Default::default()
    });

    commands.spawn_bundle(PointLightBundle {
        transform: Transform::from_xyz(4.0, 8.0, 4.0),
        ..Default::default()
    });
    commands.spawn_bundle(PerspectiveCameraBundle {
        transform: Transform::from_xyz(-2.0, 2.5, 5.0).looking_at(Vec3::ZERO, Vec3::Y),
        ..Default::default()
    });
    commands.spawn_bundle(UiCameraBundle::default());

    asset_server.watch_for_changes().unwrap();
   ```

**Comment by DJMcNab** [comment]
Viewing this model in ['3d viewer'](https://www.microsoft.com/en-gb/p/3d-viewer/9nblggh42ths#activetab=pivot:overviewtab) gives 
![image](https://user-images.githubusercontent.com/36049421/140417351-cb315b55-c0ee-4e84-bcc2-9c89b0f5bc41.png) in the 'stats and shading' tab

Looking from the back gives:
![image](https://user-images.githubusercontent.com/36049421/140417544-81a15637-5ac9-4271-b947-e498d7c266df.png)

i.e. the opposite. Therefore, your model is being correctly backface culled. I'm not sure how to disable it, unfortunately. Hopefully someone more experienced can chime in.

**Reply by VisualEhrmanntraut** [reply]
It's supposed to have a back face btw
![image](https://user-images.githubusercontent.com/30368284/140419844-07bc43ee-9f61-4167-af60-e04b53b82633.png)

**Reply by lukors** [reply]
Blender can turn on or off backface culling. Game engines often can't. A solution I often do is to duplicate the vertices and flip their normals.

**Comment by lukors** [comment, marked as answer]
I took a look at the model. Turn on backface culling in Blender and you will see the issue.

This model is not rendering incorrectly in Bevy. :)
![image](https://user-images.githubusercontent.com/1719884/140422480-32ef8534-7201-49e0-9445-d9def915f2ae.png)

**Reply by VisualEhrmanntraut** [reply]
Huh... Sorry, I'm new to 3D rendering. This stuff reads like hieroglyphs to me
